chain/src/utils/orphan_block_pool.rs

```diff
@@ -1,27 +1,28 @@
-use ckb_logger::{debug, error};
-use ckb_types::core::EpochNumber;
-use ckb_types::{core, packed};
+#![allow(dead_code)]
+use crate::LonelyBlockHash;
+use ckb_logger::debug;
+use ckb_store::{ChainDB, ChainStore};
+use ckb_types::core::{BlockView, EpochNumber};
+use ckb_types::packed;
 use ckb_util::{parking_lot::RwLock, shrink_to_fit};
 use std::collections::{HashMap, HashSet, VecDeque};
+use std::sync::Arc;
 
 pub type ParentHash = packed::Byte32;
-const SHRINK_THRESHOLD: usize = 100;
 
-// Orphan pool will remove expired blocks whose epoch is less than tip_epoch - EXPIRED_EPOCH,
-const EXPIRED_EPOCH: u64 = 6;
+const SHRINK_THRESHOLD: usize = 100;
+pub const EXPIRED_EPOCH: u64 = 6;
 
 #[derive(Default)]
 struct InnerPool {
     // Group by blocks in the pool by the parent hash.
-    blocks: HashMap<ParentHash, HashMap<packed::Byte32, core::BlockView>>,
+    blocks: HashMap<ParentHash, HashMap<packed::Byte32, LonelyBlockHash>>,
     // The map tells the parent hash when given the hash of a block in the pool.
     //
     // The block is in the orphan pool if and only if the block hash exists as a key in this map.
     parents: HashMap<packed::Byte32, ParentHash>,
     // Leaders are blocks not in the orphan pool but having at least a child in the pool.
     leaders: HashSet<ParentHash>,
```

Comment:
`block_size` is removed and method `total_size` is also removed, so the `MAX_ORPHAN_POOL_SIZE` is unused now, we should remove this constant?

Reply:
Yes

Reply:
it's introduced by this commit:
https://github.com/nervosnetwork/ckb/commit/1d4ae967c0048fb4d88ca76a99f06cb4255ec7a9

For fixing the memory issues in IBD, the code is removed in this PR, so per my understanding, that fix is only needed by the old version?

Reply:
yes